utils/message-generator/src/main.rs

```diff
@@ -17,87 +17,254 @@ use roles_logic_sv2::{common_messages_sv2::SetupConnectionSuccess, parsers::AnyM
 use std::net::SocketAddr;
 //use serde::{Deserialize as SerdeDeserialize, Serialize as SerdeSerialize};
 use serde_json;
+use tracing::{debug, info};
 
+// TODO RR: Move to `actions.rs`
+/// Represents a valid SV2 type used in the `"value"` key-pair of the `"results"` key value of the
+/// `"actions"` key value. It is used when the `"results"`'s `"type"` is `"match_message_field"`.
+/// For example:
+///    ```
+///    {
+///      "type": "match_message_field",
+///        "value": [
+///          "MiningProtocol",
+///          "OpenStandardMiningChannelSuccess",
+///          "request_id",
+///          { "U32": 89 }
+///        ]
+///     }
+///    ```
+///    where `{ "U32": 89 }` is serialized as `Sv2Type::U32(89)`.
 #[derive(Debug, PartialEq, Eq, Clone, Serialize, Deserialize)]
 enum Sv2Type {
+    /// Message field `bool` value
     Bool(bool),
+    /// Message field `u8` value
     U8(u8),
+    /// Message field `u16` value
     U16(u16),
+    /// Message field `u24` value as a vector of `u8`'s
+    /// TODO: Change variant to `U24Vec(Vec<u8>)`?
     U24(Vec<u8>),
+    /// Message field `u32` value
     U32(u32),
+    /// Message field `u256` value as a vector of `u8`'s
+    /// TODO: Change variant to `U256Vec(Vec<u8>)`?
     U256(Vec<u8>),
+    /// Message field `255` byte-len `String` value
     Str0255(String),
+    /// Message field 8-bit byte array
     B0255(Vec<u8>),
+    /// Message field 16-bit byte array
     B064K(Vec<u8>),
+    /// Message field 16-bit byte array of `u24`s
+    // TODO Q: should this be `Vec<u24>` as according to spec definition of `B0_16M`?
```

Comment:
The [`sv2-spec`](https://github.com/stratum-mining/sv2-spec/blob/main/03-Protocol-Overview.md#31-data-types-mapping) defines a `B0_16M` as "Byte array with 16-bit length prefix L. Unsigned integer encoded as **U24** above, followed by a sequence of L bytes." So should this type be `B016m(Vec<u24>)`?